src/main/java/org/elasticsearch/common/jna/Kernel32Library.java

```diff
@@ -60,6 +83,26 @@ public static Kernel32Library getInstance() {
         return Holder.instance;
     }
 
+    public boolean virtualLock(long address, long size) {
+        return internal.VirtualLock(address, size);
+    }
+
+    public int virtualQueryEx(int handle, Long address, MEMORY_BASIC_INFORMATION memoryInfo, int length) {
```

Comment:
> On Windows, you can use a Java int, since the native long type is always 32 bits.

It's 32 bits, but unsigned, so the value can exceed `Int.MAX_VALUE`.  I think it's safe to keep it as `long`.